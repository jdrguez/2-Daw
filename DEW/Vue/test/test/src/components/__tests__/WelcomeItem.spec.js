import { mount } from '@vue/test-utils';
import WelcomeItem from '../WelcomeItem.vue';
import { describe, it, expect } from "vitest";

describe('WelcomeItem.vue', () => {
  it('renders correctly with default slots', () => {
    const wrapper = mount(WelcomeItem, {
      slots: {
        icon: '<span class="test-icon">Icon</span>',
        heading: 'Test Heading',
      },
    });
    expect(wrapper.find('.item').exists()).toBe(true);
    expect(wrapper.find('.test-icon').exists()).toBe(true);
    expect(wrapper.find('h3').text()).toBe('Test Heading');
  });

  it('applies correct styles', () => {
    const wrapper = mount(WelcomeItem);
    expect(wrapper.classes()).toContain('item');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays content passed to the default slot', () => {
    const wrapper = mount(WelcomeItem, {
      slots: {
        default: '<p>Default Slot Content</p>',
      },
    });
    expect(wrapper.html()).toContain('Default Slot Content');
  });

});
