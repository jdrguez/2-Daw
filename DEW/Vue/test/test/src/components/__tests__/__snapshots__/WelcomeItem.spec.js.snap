// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WelcomeItem.vue > applies correct styles 1`] = `
<div
  class="item"
  data-v-bd9b3c35=""
>
  <i
    data-v-bd9b3c35=""
  >
    
    
  </i>
  <div
    class="details"
    data-v-bd9b3c35=""
  >
    <h3
      data-v-bd9b3c35=""
    >
      
      
    </h3>
    
    
  </div>
</div>
`;
